axelor-base/src/main/java/com/axelor/apps/base/service/LanguageServiceImpl.java

```diff
@@ -0,0 +1,197 @@
+   */
+  public long getCountMetaSelectItemFromQuery() {
+    Query<MetaSelectItem> metaSelectItemQuery =
+        metaSelectItemRepository
+            .all()
+            .filter("self.select.name = :selectName and self.select.module = :moduleName");
+    metaSelectItemQuery
+        .bind("selectName", SELECT_NAME)
+        .bind("moduleName", NEW_SELECT_LANGUAGE_MODULE);
+    return metaSelectItemQuery.count();
+  }
+
+  /**
+   * When we remove a Language obj, the related record in select.language should also be removed.
+   *
+   * @param language
+   */
+  @Transactional
+  @Override
+  public void removeLanguageLinkedMetaSelectItem(Language language) {
+    MetaSelectItem metaSelectItem = findMetaSelectItem(language.getName(), language.getCode());
+    if (metaSelectItem == null) return;
+    updateOrderNumber(metaSelectItem.getOrder());
+    metaSelectItemRepository.remove(metaSelectItem);
+  }
+
+  @Transactional
+  public void updateOrderNumber(int orderNumber) {
+    javax.persistence.Query query =
+        JPA.em()
+            .createNativeQuery(
+                "UPDATE meta_select_item "
+                    + "SET order_seq = order_seq - 1 "
+                    + "FROM meta_select "
+                    + "WHERE meta_select.id = meta_select_item.select_id "
+                    + "AND meta_select_item.order_seq > :orderNumber "
+                    + "AND meta_select.name = :selectName "
+                    + "AND meta_select.module = :moduleName ")
+            .setParameter("orderNumber", orderNumber)
+            .setParameter("selectName", SELECT_NAME)
+            .setParameter("moduleName", NEW_SELECT_LANGUAGE_MODULE);
+    query.executeUpdate();
+  }
+
+  /**
+   * Fetch the native MetaSelect and the overriddenMetaSelect. Copy MetaSelectItems from the native
+   * to the new one. If it is the first time we add a new Language object, we should create a new
+   * MetaSelect object with a name field of "select.language", a module field of "axelor-base" and a
+   * priority field of the original priority + 1. Then, we should copy all the existing
+   * MetaSelectItems from the original MetaSelect to the new MetaSelect.
+   */
+  @Transactional
+  @Override
+  public MetaSelect copyMetaSelect() {
+    // fetch the native MetaSelect
+    Query<MetaSelect> metaSelectQuery =
+        metaSelectRepository
+            .all()
+            .filter("self.name = :selectName and self.module = :nativeModuleName");
+    metaSelectQuery
+        .bind("selectName", SELECT_NAME)
+        .bind("nativeModuleName", NATIVE_SELECT_LANGUAGE_MODULE);
+    MetaSelect nativeMetaSelect = metaSelectQuery.fetchOne();
+    if (nativeMetaSelect == null) return null;
+    MetaSelect overriddenMetaSelect = getOverriddenMetaSelectFromNativeMetaSelect(nativeMetaSelect);
+    System.out.println("overriddenMetaSelect:" + overriddenMetaSelect);
+    metaSelectRepository.save(overriddenMetaSelect);
+    return overriddenMetaSelect;
```

Comment:
You must return the value of the saved overriddenMetaSelect, not just overriddenMetaSelect, so just do : 
`return metaSelectRepository.save(overriddenMetaSelect);`